scripts/scraper-ng/test/ingredient-data.interactive_example.test.js

```diff
@@ -0,0 +1,92 @@
+const { process } = require("./framework/utils");
+
+const sources = {
+  no_interactive_examples: `<p>An intro</p>
+<h2 id="A_heading">A heading</h2>
+<p>Some more stuff</p>`,
+
+  two_interactive_examples: `<p>An intro</p>
+<div>{{EmbedInteractiveExample()}}</div>
+<p>Some more stuff</p>
+<div>{{EmbedInteractiveExample()}}</div>
+<h2 id="A_heading">A heading</h2>
+<p>Some more stuff</p>`,
+
+  interactive_example_after_h2: `<p>An intro</p>
+<h2 id="A_heading">A heading</h2>
+<p>Some more stuff</p>
+<div>{{EmbedInteractiveExample()}}</div>`,
+
+  interactive_example_not_in_div: `<p>An intro</p>
+<p>{{EmbedInteractiveExample()}}</p>
+<h2 id="A_heading">A heading</h2>
+<p>Some more stuff</p>`,
+
+  interactive_example_not_after_p: `<div>An intro</div>
+<div>{{EmbedInteractiveExample()}}</div>
+<h2 id="A_heading">A heading</h2>
+<p>Some more stuff</p>`,
+
+  valid_interactive_example: `<p>An intro</p>
+<div>{{EmbedInteractiveExample()}}</div>
+<h2 id="A_heading">A heading</h2>
+<p>Some more stuff</p>`,
+};
+
+describe("data.examples", () => {
+  const testRecipe = { body: ["data.interactive_example?"] };
+
+  test("no interactive examples", () => {
+    const file = process(sources.no_interactive_examples, testRecipe);
+
+    expect(file.messages.length).toBe(1);
+    expect(file).hasMessageWithId("/unknown-heading");
```

Comment:
Why do these tests assert the unknown heading message? Since they exercise another rule, I'm not sure how they increase our confidence in this ingredient's handler. How about asserting that it _doesn't_ have a `data.interactive_example?` message? Something like…

```js
    expect(file).not.hasMessageWithId("data.interactive_example?");
```

Reply:
This is tricky though where there are actual interactive example errors. What I want to say, I suppose, is: there are _only these particular_ `data.interactive_example?` errors, and I have tried to cobble together a regex to that effect.